app/routes.py

```diff
@@ -1,2 +1,268 @@
-from flask import Blueprint
+from types import new_class 
+import requests
+from app import db
+from flask import Blueprint, jsonify, request
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import os
+
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["POST", "GET"])
+def handle_tasks():
+    # POST REQUESTS
+    if request.method == "POST":
+
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            
+            return jsonify({"details": "Invalid data"}), 400
+
+        new_task = Task(
+            title = request_body["title"],
+            description = request_body["description"]
+        )
+        if "completed_at" in request_body:
+            new_task.completed_at = request_body["completed_at"]
+
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        new_task_response = { 
+            "task": new_task.get_task_dict()
+        }
+
+        return jsonify(new_task_response), 201
+
+    # GET REQUESTS
+    elif request.method == "GET":
+        title_query = request.args.get("title")
+        description_query = request.args.get("description")
+        if title_query:
```

Comment:
if you would like to combine the title and sort query you could create one if/elif statement like this 
`if not sort_method:
        tasks = Task.query.all()
    elif sort_method == "asc":
        tasks = Task.query.order_by(Task.title.asc()).all()`